selfdrive/car/honda/carstate.py

```diff
@@ -263,6 +262,21 @@ def update(self, cp, cp_cam, cp_body):
     ret.cruiseState.enabled = cp.vl["POWERTRAIN_DATA"]["ACC_STATUS"] != 0
     ret.cruiseState.available = bool(cp.vl[self.main_on_sig_msg]["MAIN_ON"])
 
+    steer_status = self.steer_status_values[cp.vl["STEER_STATUS"]["STEER_STATUS"]]
+    # LOW_SPEED_LOCKOUT is not worth a warning on most cars
+    # NO_TORQUE_ALERT_2 can be caused by bump or steering nudge from driver
+    ret.steerFaultPermanent = steer_status not in ("NORMAL", "NO_TORQUE_ALERT_1", "NO_TORQUE_ALERT_2", "LOW_SPEED_LOCKOUT", "TMP_FAULT")
+    if self.CP.minSteerSpeed > 0.:
+      # TODO: make generic for all makes and clean up
+      # Check for steer mismatch when engaged and moving before setting a temporary fault. Allow for a small buffer as to not flash the alert when engaging.
+      self.steer_mismatch_cnt = self.steer_mismatch_cnt + 1 if ret.cruiseState.enabled and \
+                                not bool(cp.vl["STEER_STATUS"]["STEER_CONTROL_ACTIVE"]) and not ret.standstill else 0
+      steer_mismatch = True if self.steer_mismatch_cnt > MAX_STEPS_STEER_MISMATCH else False
+    else:
+      steer_mismatch = False
+    # TODO: detect the torque request diff from OP vs the radar
```

Comment:
it might be fine to just let the steer saturated logic handle large request differences